proposals/20210525-feast_example_gen.md

```diff
@@ -0,0 +1,102 @@
+#### SIG TFX-Addons
+# Project Proposal
+
+**Your name:** Badrul Chowdhury
+
+**Your email:** badrulchowdhury17@gmail.com
+
+**Your company/organization:** Apple
+
+**Project name:** ExampleGen for Feast Feature Store
+
+## Project Description
+
+This proposal is for a new component to enable connecting to and reading from feature stores via a simple and unified API. The first iteration will only support Feast [1], but we will add support for other popular feature stores over time. Similarly, we will restrict v1 of our library to only read from feature stores- we will evaluate supporting writes later.
+
+## Project Category
+Component
+
+## Project Use-Case(s)
+
+Feature stores such as Feast are becoming increasingly popular as (managed) platforms to manage all feature-related information. A separate feature store has several benefits including (but not limited to) the following [2]:
+
+1. Feature sharing and reuse
+2. Serving features at scale
+3. Consistency between training and serving (inference)
+4. Point-in-time correctness
+5. Data quality and validation
+
+Different organizations may use different feature stores in their ML stack, each with its own APIs for reading online and historical features. This is a significant barrier to users as they will have to learn to use a different set of APIs for each feature store. Furthermore, different feature stores may return different feature vector representations. The proposed component will solve these issues:
+
+1. It will provide a unified API across different feature stores. It will hide the mechanics of reading from different feature stores from the user.
+2. It will always return the feature vector as a tensor. It will convert different representations of the feature vector to a tensor transparently from the user.
```

Comment:
(assuming this will be implemented as an ExampleGen)

The component still needs to output tensors into serialized formats on disk. Do you foresee tf.Example suffices as tensor container format, or we might need to support more formats?

non blocking, just asking for completeness